lib/brainstem/api_docs/formatters/open_api_specification/version_2/presenter_formatter.rb

```diff
@@ -52,18 +52,45 @@ def format_type!
             end
 
             def format_fields!
-              return unless presenter.valid_fields.any?
+              return unless presenter.valid_fields.any? || presenter.valid_associations.any?
 
-              definition.merge! properties: format_field_branch(presenter.valid_fields)
+              properties = format_field_branch(presenter.valid_fields)
+              with_associations = format_field_associations(properties)
+
+              definition.merge! properties: with_associations
             end
 
             def format_field_branch(branch)
-              branch.inject(ActiveSupport::HashWithIndifferentAccess.new) do |buffer, (name, field)|
+              branch.each_with_object(ActiveSupport::HashWithIndifferentAccess.new) do |(name, field), buffer|
                 buffer[name.to_s] = format_field(field)
-                buffer
               end
             end
-            
+
+            def format_field_associations(properties)
+              presenter.valid_associations.each_with_object(properties) do |(_name, association), props|
+                if association.foreign_key
+                  case association.type
+                  when :belongs_to, :has_one
+                    props[association.foreign_key] = type_and_format(:integer).merge(description: association.description) unless props[association.foreign_key]
+                  when :has_many
```

Comment:
> Rails supports six types of associations:
> - belongs_to
> - has_one
> - has_many
> - has_many :through
> - has_one :through
> - has_and_belongs_to_man

We might be missing three of these.

Reply:
I see that we're only intentionally support three. Is this for the time being?

Reply:
The type specified on the association is meant to determine the key name for the association.

In terms of having dynamic keys on the presented object, these six associations fall in two buckets. For `belongs_to`, `has_one ` & `has_one :through` will generate attribute name of `blah_id` whereas all the other ones will generate `blah_ids`